Convert K&R function definitions to ANSI prototypes.

getDateWeightFromJulian(), getSimpleTdefsByNumber(), and
getTdefFunctionsByNumber() used pre-C89 parameter lists with
implicit-int parameters.  GCC 14 makes implicit int a hard error by
default.  Also include <strings.h> where strcasecmp() is used so the
prototype is visible under strict dialects.  No behavior change.

Index: dbt7/dsgen/tools/date.c
===================================================================
--- dbt7.orig/dsgen/tools/date.c
+++ dbt7/dsgen/tools/date.c
@@ -563,7 +563,7 @@ day_number(date_t *d)
  * TODO: None
  */
 int
-getDateWeightFromJulian(jDay, nDistribution)
+getDateWeightFromJulian(int jDay, int nDistribution)
 {
 	date_t dTemp;
 	int nDay;
Index: dbt7/dsgen/tools/tdefs.c
===================================================================
--- dbt7.orig/dsgen/tools/tdefs.c
+++ dbt7/dsgen/tools/tdefs.c
@@ -36,6 +36,7 @@
 #include "config.h"
 #include "porting.h"
 #include <stdio.h>
+#include <strings.h>
 #include "tables.h"
 #include "columns.h"
 #include "genrand.h"
@@ -147,7 +148,7 @@ getTdefsByNumber(int nTable)
 }
 */
 tdef *
-getSimpleTdefsByNumber(nTable)
+getSimpleTdefsByNumber(int nTable)
 {
    if (nTable >= S_BRAND)
       return(&s_tdefs[nTable - S_BRAND]);
Index: dbt7/dsgen/tools/tdef_functions.c
===================================================================
--- dbt7.orig/dsgen/tools/tdef_functions.c
+++ dbt7/dsgen/tools/tdef_functions.c
@@ -171,7 +171,7 @@ table_func_t s_tdef_funcs[] = {
 };
 
 table_func_t *
-getTdefFunctionsByNumber(nTable)
+getTdefFunctionsByNumber(int nTable)
 {
    if (nTable >= S_BRAND)
       return(&s_tdef_funcs[nTable - S_BRAND]);
